application/src/main/java/org/togetherjava/tjbot/commands/BotCommandAdapter.java

```diff
@@ -136,10 +137,13 @@ protected final String generateComponentId(Lifespan lifespan, String... args) {
     }
 
     /**
-     * @return the component id generator used by {@link #generateComponentId(String...)} and
-     *         {@link #generateComponentId(Lifespan, String...)}
+     * Gets the generator used to create component IDs.
+     * <p>
+     * In general, prefer using {@link #generateComponentId(Lifespan, String...)} and {@link #generateComponentId(String...)} instead of interacting with the generator directly.
+     *
+     * @return the generator
      */
-    protected final ComponentIdGenerator getComponentIdGenerator() {
+    protected final @Nullable ComponentIdGenerator getComponentIdGenerator() {
```

Comment:
the method is **not** supposed to return a nullable.

its a mistake to call the method before intiialization, not allowed. the method should fail-fast if its called before.